cryptography/src/bls12381/scheme.rs

```diff
@@ -98,3 +103,521 @@ impl Scheme for Bls12381 {
+        )
+    }
+
+    // sign_case_37286e1a6d1f6eb3
+    fn vector_sign_5() -> (Vec<u8>, Vec<u8>, Vec<u8>) {
+        vector_sign(
+        "0x47b8192d77bf871b62e87859d653922725724a5c031afeabc60bcef5ff665138",
+        "0xabababababababababababababababababababababababababababababababab",
+        "0x9674e2228034527f4c083206032b020310face156d4a4685e2fcaec2f6f3665aa635d90347b6ce124eb879266b1e801d185de36a0a289b85e9039662634f2eea1e02e670bc7ab849d006a70b2f93b84597558a05b879c8d445f387a5d5b653df"
+        )
+    }
+
+    // sign_case_7055381f640f2c1d
+    fn vector_sign_6() -> (Vec<u8>, Vec<u8>, Vec<u8>) {
+        vector_sign(
+        "0x328388aff0d4a5b7dc9205abd374e7e98f3cd9f3418edb4eafda5fb16473d216",
+        "0x0000000000000000000000000000000000000000000000000000000000000000",
+        "0x948a7cb99f76d616c2c564ce9bf4a519f1bea6b0a624a02276443c245854219fabb8d4ce061d255af5330b078d5380681751aa7053da2c98bae898edc218c75f07e24d8802a17cd1f6833b71e58f5eb5b94208b4d0bb3848cecb075ea21be115"
+        )
+    }
+
+    // sign_case_c82df61aa3ee60fb
+    fn vector_sign_7() -> (Vec<u8>, Vec<u8>, Vec<u8>) {
+        vector_sign(
+        "0x263dbd792f5b1be47ed85f8938c0f29586af0d3ac7b977f21c278fe1462040e3",
+        "0x0000000000000000000000000000000000000000000000000000000000000000",
+        "0xb6ed936746e01f8ecf281f020953fbf1f01debd5657c4a383940b020b26507f6076334f91e2366c96e9ab279fb5158090352ea1c5b0c9274504f4f0e7053af24802e51e4568d164fe986834f41e55c8e850ce1f98458c0cfc9ab380b55285a55"
+        )
+    }
+
+    // sign_case_d0e28d7e76eb6e9c
+    fn vector_sign_8() -> (Vec<u8>, Vec<u8>, Vec<u8>) {
+        vector_sign(
+        "0x263dbd792f5b1be47ed85f8938c0f29586af0d3ac7b977f21c278fe1462040e3",
+        "0x5656565656565656565656565656565656565656565656565656565656565656",
+        "0x882730e5d03f6b42c3abc26d3372625034e1d871b65a8a6b900a56dae22da98abbe1b68f85e49fe7652a55ec3d0591c20767677e33e5cbb1207315c41a9ac03be39c2e7668edc043d6cb1d9fd93033caa8a1c5b0e84bedaeb6c64972503a43eb"
+        )
+    }
+
+    // sign_case_f2ae1097e7d0e18b
+    fn vector_sign_9() -> (Vec<u8>, Vec<u8>, Vec<u8>) {
+        vector_sign(
+        "0x263dbd792f5b1be47ed85f8938c0f29586af0d3ac7b977f21c278fe1462040e3",
+        "0xabababababababababababababababababababababababababababababababab",
+        "0x91347bccf740d859038fcdcaf233eeceb2a436bcaaee9b2aa3bfb70efe29dfb2677562ccbea1c8e061fb9971b0753c240622fab78489ce96768259fc01360346da5b9f579e5da0d941e4c6ba18a0e64906082375394f337fa1af2b7127b0d121"
+        )
+    }
+
+    // sign_case_zero_privkey
+    fn vector_sign_10() -> (Vec<u8>, Vec<u8>, Vec<u8>) {
+        vector_sign(
+            "0x0000000000000000000000000000000000000000000000000000000000000000",
+            "0xabababababababababababababababababababababababababababababababab",
+            "",
+        )
+    }
+
+    // verify_infinity_pubkey_and_infinity_signature
+    fn vector_verify_1() -> (Vec<u8>, Vec<u8>, Vec<u8>, bool) {
```

Comment:
nit: It might be cleaner to use the same underlying parsing for both sign/verify and for verify to just add `bool` outside the function to the returned tuple. Right now, you are basically passing in the bool to get it right back.